support/manager.go

```diff
@@ -181,7 +181,7 @@ func materializeFlow(flowRep *definition.DefinitionRep) (*definition.Definition,
 
 	def, err := definition.NewDefinition(flowRep)
 	if err != nil {
-		return nil, fmt.Errorf("error unmarshalling flow: %s", err.Error())
+		return nil, err
```

Comment:
Any particular reason to remove "error unmarshalling" ? May be we can add "Error creating flow" since it's not unmarshalling but rather creating the Flow Def., thoughts ?

Reply:
Thanks for your comment @skothari-tibco, From the end-user, they don't care about the details step and they care about the error message can clear let them know what wrong. 
 we reporting all error messages inside NewDefinition. 

I think we can remove it and  `error unmarshalling` something that confuses the user and not that cleaner

The purpose to try to make the overall error message looks more clear

Reply:
A end-user can be a contributor/developer too. So I think that would be convenient and faster to debug.

Reply:
@skothari-tibco  Yes, it can be a developer too.  I am thinking do we really need that?

Failed to create engine: creatint task [log_3] in flow [subflow] error : xxxx
  vs
Failed to create engine: error creating flow: creating task [log_3] in flow [subflow] error: xxxx

As long as we report every error clearly when creating flow definition I think that should good then.  @fm-tibco thoughts?

Reply:
I would make sense if we had the flow name at the "unmarshal" level (would avoid the nested messages having to say "in flow [name]" but given that we don't... I'm not sure not sure. `Failed to create engine: failed to create flow:  error creating task[name] in flow[name] : ` does look a little odd.  @mellistibco any thoughts on this?

Reply:
@fm-tibco How about we log more technical description at debug level?
For error log where flow name is not available,  ```Failed to create engine: failed to load flow due to configuration issue in task[name] ```.

Reply:
I guess we could propagate error as `error creating task [log_3] in the flow [subflow]` and debug log `error creating task [log_3] in the flow [subflow]: ...error details...`.  (I know that its a bit redundant including the original message.. but will be easier to correlate that way).

@mellistibco any thoughts on this?

Reply:
I was also thinking about the schema stuff that we have in that function. 
I agree `Failed to create engine: creating task [log_3] in flow [subflow] error : xxxx` let's us know the error precisely, but with the schema I thought it would be helpful having "error creating flow:" . @fm-tibco thoughts ?

Reply:
@skothari-tibco  good catch for schema stuff. I missed adding error for schemas.  I will add similar stuff.
```
					return nil, fmt.Errorf("unable to find/create input [%s]'s schema [%+v]:%s", name, def, err.Error())

```

Let's wait for @mellistibco thoughts on `error creating flow` :)

Reply:
to be honest, I think its just annoying to hide the real cause of an error. I agree that `unmarshling` doesn't really do it, but we should be as verbose as possible to help find the issue rather than just obscuring it so that I'd need to enable debug logging and try again.

As for the actual error message, what about something like `failed to create engine: error [parsing || creating] task [whatever] in flow [blah]: [the actual error, within reason, just enough to show the problem but not too much to confuse someone less technical]`

Reply:
@mellistibco Agree, we are trying to make the error message easier to understand also easy to catch the real problem.   

Are you ok with below error message？
```
 Failed to create engine: error creating task [log_3] in flow [subflow]: unable to convert input [addDetails]'s value [asdasd] to type [bool]:strconv.ParseBool: parsing "asdasd": invalid syntax
```

Reply:
@lixingwang works for me if everyone else is ok with it...

Reply:
Thanks, @mellistibco,   @fm-tibco @vijaynalawade @skothari-tibco :  Please approve PR, if you are ok with the above error message, thx.